<p>This rule reminds team members to use BillPugh implementation when using singleton.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    <code>
public final class SingletonExample {

    private static SingletonExample INSTANCE;

    private SingletonExample() {
    }

    public static SingletonExample getInstance() {
        if(INSTANCE == null) {
            INSTANCE = new SingletonExample();
        }

        return INSTANCE;
    }

    // getters and setters
}
    </code>
</pre>
<h2>Compliant Solution</h2>
<pre>
    <code>
public class SingletonExample {
    private SingletonExample() {

    }

    private static class SingletonHelper {
        private static final SingletonExample BILL_PUGH_SINGLETON_INSTANCE = new SingletonExample();
    }

    public static SingletonExample getInstance() {
        return SingletonHelper.BILL_PUGH_SINGLETON_INSTANCE;
    }
}
    </code>
</pre>
